SU2_CFD/src/iteration_structure.cpp

```diff
@@ -2588,16 +2586,14 @@ bool CDiscAdjFluidIteration::Monitor(COutput *output,
 
   /*--- Write the convergence history for the fluid (only screen output) ---*/
 
-  output->SetHistory_Output(geometry[ZONE_0][INST_0][MESH_0], 
-                            solver[ZONE_0][INST_0][MESH_0], 
-                            config[ZONE_0], 
-                            config[ZONE_0]->GetTimeIter(),
-                            config[ZONE_0]->GetOuterIter(), 
-                            config[ZONE_0]->GetInnerIter());
-  
-  StopCalc = output->GetConvergence();
-  
-  return StopCalc;
+  output->SetHistory_Output(geometry[val_iZone][INST_0][MESH_0],
+                            solver[val_iZone][INST_0][MESH_0],
+                            config[val_iZone],
+                            config[val_iZone]->GetTimeIter(),
+                            config[val_iZone]->GetOuterIter(),
+                            config[val_iZone]->GetInnerIter());
+
+  return output->GetConvergence();
```

Comment:
@talbring I could use your help with something, in some CIteration methods there is a lot of hardcoded ZONE_0 even when `iZone` is passed as argument, this particular case was causing segfaults when ZONE_0 was not a fluid zone.
Maybe there are others but I don't want to change them blindly, is there any rule for when to use ZONE_0 / INST_0 instead of iZone / iInst ?

Reply:
whenever there is `val_iZone` and/or `val_iInst` passed as an argument it should be used in that function ...

Reply:
Of course we could also think of just passing the correct pointers. Maybe that would be even better, since the iteration classes should only deal with individual zones. Will be part of a future restructuring of the CIteration structure.